perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -902,6 +919,49 @@ public RichMember getRichMemberWithAttributes(PerunSession sess, Member member)
 		return richMembers;
 	}
 
+	public List<RichMember> convertMembersToRichMembersWithAttributes(PerunSession sess, List<RichMember> richMembers, Resource resource, List<AttributeDefinition> attrsDef)  throws InternalErrorException, WrongAttributeAssignmentException {
+		Facility facility = getPerunBl().getResourcesManagerBl().getFacility(sess, resource);
+		List<AttributeDefinition> usersAttributesDef = new ArrayList<AttributeDefinition>();
+		List<AttributeDefinition> membersAttributesDef = new ArrayList<AttributeDefinition>();
+		List<AttributeDefinition> userFacilityAttributesDef = new ArrayList<AttributeDefinition>();
+		List<AttributeDefinition> memberResourceAttributesDef = new ArrayList<AttributeDefinition>();
+
+		List<String> attrNames = new ArrayList<>();
+		for(AttributeDefinition attributeDefinition: attrsDef) {
+			attrNames.add(attributeDefinition.getName());
+		}
+
+
+		for(AttributeDefinition attrd: attrsDef) {
+			if(attrd.getName().startsWith(AttributesManager.NS_USER_ATTR)) usersAttributesDef.add(attrd);
+			else if(attrd.getName().startsWith(AttributesManager.NS_USER_FACILITY_ATTR)) userFacilityAttributesDef.add(attrd);
+			else if(attrd.getName().startsWith(AttributesManager.NS_MEMBER_ATTR)) membersAttributesDef.add(attrd);
+			else if(attrd.getName().startsWith(AttributesManager.NS_MEMBER_RESOURCE_ATTR)) membersAttributesDef.add(attrd);
+		}
+
+		for (RichMember richMember: richMembers) {
+			List<Attribute> userAttributes = new ArrayList<Attribute>();
+			List<Attribute> memberAttributes = new ArrayList<Attribute>();
+
+			List<Attribute> attributes = getPerunBl().getAttributesManagerBl().getAttributes(sess, resource, richMember, attrNames, true);
+
+			for(Attribute attribute: attributes) {
+				if(attribute.getNamespace().startsWith("urn:perun:user")) {
```

Comment:
Since above getAttributes() method returns also member_resource and user_facility attributes, you store them in same property as member/user attributes. I'm not fond with this, but we probably don't want to extend current RichMember object.